scripts/ccpp_datafile.py

```diff
@@ -288,6 +290,23 @@ def _retrieve_dependencies(table):
     # end for
     return sorted(result)
 
+###############################################################################
+def _retrieve_dyn_const_routines(table):
+###############################################################################
+    """Find and return a list of all scheme constituent routines."""
+    result = set()
+    routines = table.find("dyn_const_routines")
+    if routines is None:
+        raise CCPPDatatableError("Could not find 'dyn_const_routine' element")
```

Comment:
Not to be pedantic but I think we are missing an `s` here:

```suggestion
        raise CCPPDatatableError("Could not find 'dyn_const_routines' element")
```

Reply:
fine. fixed.